### PR TITLE
Fix encrypting streams

### LINKED ISSUE #89: Can't encrypt stream objects (StringIO, BytesIO, etc.)
**POC**

```py
$ python
Python 2.7.6 (default, Mar 22 2014, 22:59:56)
>>> import gnupg
>>> gpg = gnupg.GPG()
>>> plaintext = "No justice, no peace"
>>> type(plaintext)
<type 'str'>
>>> out = gpg.encrypt(plaintext, symmetric=True, passphrase="no racist police", encrypt=False)
>>> out.ok
True
>>> from StringIO import StringIO
>>> plaintext = StringIO("No justice, no peace")
>>> out = gpg.encrypt(plaintext, symmetric=True, passphrase="no racist police", encrypt=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/gnupg/gnupg.py", line 955, in encrypt
    stream = _make_binary_stream(data, self._encoding)
  File "/usr/local/lib/python2.7/dist-packages/gnupg/_util.py", line 396, in _make_binary_stream
    s = s.encode(encoding)
AttributeError: StringIO instance has no attribute 'encode'
```

There are numerous references to files and file-like objects in the docstrings of these functions, which leads me to believe this is supposed to work. It did work when using `encrypt_file` in 1.2.5, but it was broken when encrypt_file was changed to be `encrypt` in 295d98fbdc6a9bff927da8fe011241d7bbbd8a3c.

A closer look at the site of the traceback shows:

1. If the input to `encrypt` is a byte string (definition of that depends on Python 2.x or 3.x), it is wrapped in either a BytesIO or a StringIO and returned.
2. Otherwise, it is first encoded (default is `utf-8`), then converted into a stream object and returned.

Therefore, it seems that if `data` is *already* a stream object, we should simply return it rather than trying (and failing) to encode it - which doesn't really make sense for a stream object anyway.

The most obvious fix would be something like:

```py
            if type(s) not in (str, StringIO, BytesIO): # etc.
                s = s.encode(encoding)
```

Does that seem reasonable, or am I missing something?

### PR DESCRIPTION
**Pull request by garrettr**
Fixes #89.

### CODE REVIEW

File: gnupg/_util.py [hunk 1]
```diff
@@ -41,6 +41,17 @@
 except ImportError:
     from cStringIO import StringIO
 
+# Import the StringIO class from the StringIO module since it is a
+# commonly used stream class. It is distinct from either of the
+# StringIO's that may be loaded in the above try/except clause, so the
+# name is prefixed with an underscore to distinguish it.
+from StringIO import StringIO as _StringIO
```
Comment (ttanner) [inline gnupg/_util.py]: this fail on Python 3

Comment (garrettr) [inline gnupg/_util.py]: @ttanner Thanks for testing that! I figured the test might need some additional conditionals to work properly on Python 3, but wanted to wait to get the go-ahead that this overall approach was indeed the right one before investing any more time in this PR.

**Comment by garrettr** [comment]
@isislovecruft Any thoughts on this?

**Comment by isislovecruft** [comment]
@garrettr It's great! I'm fixing the Python3 import issues that @tanner caught, and then it'll be merged for the next release (2.0.1). Thanks!

**Pull request merged**